dattri/algorithm/influence_function.py

```diff
@@ -815,3 +815,187 @@ def _get_layer_wise_reps(self,
+
+        Args:
+            full_train_dataloader (DataLoader): The dataloader
+                with full training samples for inverse hessian calculation.
+            max_iter (int, optional): An integer indicating the maximum number of
+                batches that will be used for estimating the the covariance matrices
+                and lambdas.
+        """
+        from dattri.func.fisher import (
+            estimate_covariance,
+            estimate_eigenvector,
+            estimate_lambda,
+        )
+
+        self._set_full_train_data(full_train_dataloader)
+
+        if max_iter is None:
+            max_iter = len(full_train_dataloader)
+
+        func = partial(self.task.get_target_func(), self.task.get_param()[0])
+        # 1. Use random batch to estimate covariance matrices S and A
+        cov_matrices = estimate_covariance(func,
+                                           full_train_dataloader,
+                                           self.layer_cache,
+                                           max_iter,
+                                           device=self.device)
+
+        # 2. Calculate the eigenvalue decomposition of S and A
+        self.cached_q = estimate_eigenvector(cov_matrices)
+
+        # 3. Use random batch for eigenvalue correction
+        self.cached_lambdas = estimate_lambda(func,
+                                              full_train_dataloader,
+                                              self.cached_q,
+                                              self.layer_cache,
+                                              max_iter,
+                                              device=self.device)
+
+        # Remove hooks after preprocessing the FIM
+        for handle in self.handles:
+            handle.remove()
+
+    def transform_test_rep(
+        self,
+        ckpt_idx: int,
+        test_rep: torch.Tensor,
+    ) -> torch.Tensor:
+        """Calculate the transformation on the test representations.
+
+        Args:
+            ckpt_idx (int): Index of the model checkpoints. Used for ensembling
+                different trained model checkpoints.
+            test_rep (torch.Tensor): Test representations to be transformed.
+                Typically a 2-d tensor with shape (batch_size, num_parameters).
+
+        Returns:
+            torch.Tensor: Transformed test representations. Typically a 2-d
+                tensor with shape (batch_size, transformed_dimension).
+
+        Raises:
+            ValueError: If specifies a non-zero `ckpt_idx`.
+        """
+        if ckpt_idx != 0:
+            error_msg = ("EK-FAC only supports single model checkpoint, "
+                         "but receives non-zero `ckpt_idx`.")
+            raise ValueError(error_msg)
+
+        # Unflatten the test_rep
+        full_model_params = {
+            k: p for k, p in self.task.model.named_parameters() if p.requires_grad
+        }
+        partial_model_params = {
```

Comment:
How about using self.task.get_param(layer_name=self.layer_name, layer_split=True)

Reply:
I think it's a bit hard to use this function, because it only provides the flattened parameters? Here we need the original shape information for each layer.

Reply:
I think layer_split = True will give you a map to the module name.

Maybe @TheaperDeng has better idea here.

Reply:
Since I think this can be done by easily change to the `get_param`. So I think I can handle this in next PR, and leave it as it is now in this PR.